lib/OpenQA/Schema/Result/Comments.pm

```diff
@@ -132,9 +132,17 @@ Returns C<build_nr>, C<type> and optionally C<description> if C<$self> is tag,
 e.g. 'tag:0123:important:GM' returns a list of '0123', 'important' and 'GM'.
 =cut
 sub tag ($self) {
-    $self->text
-      =~ /\btag:((?<version>[-.@\d\w]+)-)?(?<build>[-.@\d\w]+):(?<type>[-@\d\w]+)(:(?<description>[-.@\d\w]+))?\b/;
-    return $+{build}, $+{type}, $+{description}, $+{version};
+    my $text = $self->text;
+    if ($text
+        =~ /\btag:((?<version>[-.@\d\w]+)-)?(?<build>[-.@\d\w]+):(?<type>[-@\d\w]+)(:(?<description>[-.@\d\w]+))?\b/)
+    {
+        return $+{build}, $+{type}, $+{description}, $+{version};
+    }
+    elsif ($text
+        =~ /\btag:"((?<version>[-.@\d\w]+)-)?(?<build>[-.@\d\w\s\+:]+)":(?<type>[-@\d\w]+)(:(?<description>[-.@\d\w]+))?\b/)
+    {
+        return $+{build}, $+{type}, $+{description}, $+{version};
+    }
```

Comment:
It would be nice if the `return` paths could be de-duplicated. Maybe the following works?
```suggestion
    if (($text
        =~ /\btag:((?<version>[-.@\d\w]+)-)?(?<build>[-.@\d\w]+):(?<type>[-@\d\w]+)(:(?<description>[-.@\d\w]+))?\b/)
        || ($text
        =~ /\btag:"((?<version>[-.@\d\w]+)-)?(?<build>[-.@\d\w\s\+:]+)":(?<type>[-@\d\w]+)(:(?<description>[-.@\d\w]+))?\b/))
    {
        return $+{build}, $+{type}, $+{description}, $+{version};
    }
```